app/uk/gov/hmrc/tai/service/TaxCodeChangeService.scala

```diff
@@ -59,16 +59,14 @@ class TaxCodeChangeService @Inject()(taxCodeChangeConnector: TaxCodeChangeConnec
 
   def lastTaxCodeRecordsInYearPerEmployment(nino: Nino, year: TaxYear)(
     implicit hc: HeaderCarrier): Future[List[TaxCodeRecord]] =
-    taxCodeChangeConnector.lastTaxCodeRecords(nino, year) map {
-      case TaiSuccessResponseWithPayload(taxCodeRecords: List[TaxCodeRecord]) => taxCodeRecords
-      case TaiTaxAccountFailureResponse(_) =>
-        throw new RuntimeException(s"Could not fetch last tax code records for year $year")
-    }
+    taxCodeChangeConnector.lastTaxCodeRecords(nino, year)
 
   def hasTaxCodeRecordsInYearPerEmployment(nino: Nino, year: TaxYear)(implicit hc: HeaderCarrier): Future[Boolean] =
-    taxCodeChangeConnector.lastTaxCodeRecords(nino, year) map {
-      case TaiSuccessResponseWithPayload(taxCodeRecords: List[TaxCodeRecord]) if taxCodeRecords.nonEmpty => true
-      case _                                                                                             => false
+    taxCodeChangeConnector.lastTaxCodeRecords(nino, year).map {
+      case taxCodeRecords: List[TaxCodeRecord] if taxCodeRecords.nonEmpty => true
```

Comment:
u could just do `taxCodeChangeConnector.lastTaxCodeRecords(nino, year).map(_.nonEmpty)`

Reply:
`taxCodeChangeConnector.lastTaxCodeRecords(nino, year).attemptT.fold(_ => false, _.nonEmpty)` or something like this